src/back/glsl.rs

```diff
@@ -899,17 +1163,31 @@ fn write_array_size(size: ArraySize) -> Result<String, Error> {
     })
 }
 
-fn write_image_flags(flags: ImageFlags) -> Result<String, Error> {
+fn write_image_flags(flags: ImageFlags, features: SupportedFeatures) -> Result<String, Error> {
     let mut out = String::new();
```

Comment:
another way to write this (and that's how MSL backend does it) would be:
```
let ms = if flags.contains(ImageFlags::MULTISAMPLED) {.. } else { "" };
let array = if flags.contains(ImageFlags::ARRAYED) {"Array"} else { "" };
Ok(format!("{}{}", ms, array))
```
This has the benefit of not using the `write!` on a string, since it may be a bit confusing